vortex-serde/src/layouts/read/mask.rs

```diff
@@ -95,6 +95,16 @@ impl RowMask {
         })
     }
 
+    pub fn with_values(self, values: Array) -> VortexResult<Self> {
+        // TODO(robert): Avoid densifying sparse values just to get true indices
+        let sparse_mask =
+            SparseArray::try_new(self.to_indices_array()?, values, self.len(), false.into())?
+                .into_array()
+                .into_bool()?
+                .into_array();
```

Comment:
This canonicalizes immediately since we don't have bool iterators for sparse arrays